tools/manual/cp2k_input.xsl

```diff
@@ -72,6 +72,12 @@
      There can be multiple variable statements per line. The expansion process is repeated
      until no more variables are found.
     </dd>
+    <dt><b>${VAR-DEFAULT}</b></dt>
```

Comment:
This syntax can lead to ambiguities because variable names can contain pretty much any character. The only save choice for a separator is probably whitespace, ie. `${VAR default}`. Alternatively, we could use the bash syntax which uses two chars `:-`. Furthermore, this feature should probably be disabled for variables without braces, ie. `$VAR`.

Reply:
> This syntax can lead to ambiguities because variable names can contain pretty much any character. The only save choice for a separator is probably whitespace, ie. `${VAR default}`. 

Mildly. If someone wants to use default vars he can't use dashes anymore.
But yes, it might be a good idea to restrict the character set for variable names.

> Alternatively, we could use the bash syntax which uses two chars `:-`.

It is [bash syntax](https://www.tldp.org/LDP/abs/html/parameter-substitution.html), `:-` has a slightly different behavior I didn't want to implement (yet): using the default value if the content of the var is empty.
Besides, it doesn't actually solve the ambiguity, just makes it a bit less likely.

> Furthermore, this feature should probably be disabled for variables without braces, ie. `$VAR`.

It already is only available for vars with braces.

Reply:
The cleanest solution would probably be `@SETDEFAULT`, still we shouldn't do it.

Reply:
btw, this extension solves the issue of providing XC includes with default values

Reply:
As discussed in https://github.com/cp2k/cp2k/pull/559 we shouldn't really be using the preprocessor for canned XC sections. Instead we should build on what we use e.g. for [PBE0](https://github.com/cp2k/cp2k/blob/master/src/input_cp2k_check.F#L154). That solution is clean, robust, extendable, and more user-friendly.

Btw. I'm marking this as unresolved, because my original concerns about ambiguities and backwards incompatibility have not yet been address.

Reply:
OK, restriction added for the variable names. With the example given in the documentation we have already implied that the variable name is a word, so I don't see a problem of defining the supported character set now (also, `$` and ` ` were already not part of the set).